src/storage/src/hummock/state_store.rs

```diff
@@ -247,8 +247,9 @@ impl StateStore for HummockStorage {
                             .table_infos
                             .partition_point(|table| {
                                 let ord = VersionedComparator::compare_key(
-                                    user_key(&table.key_range.as_ref().unwrap().left),
-                                    key,
+                                    &FullKey::from_slice(&table.key_range.as_ref().unwrap().left)
+                                        .into_inner(),
+                                    &FullKey::from_user_key_slice(key, epoch).into_inner(),
```

Comment:
Still strange. We need to construct a FullKey and immediately consume it. A lot of copy here

Reply:
I'd like to have a `FullKeySlice` structure here. https://github.com/dtolnay/ref-cast may help to convert &[u8] into a `FullKeySlice`.

Reply:
I think it do not works. Refcast only accept one parameters like &[u8]. However, we get epoch and userkey (&[u8]). No idea how to concatenate them..